File: frontend/src/service/achievementService.ts
import api from "../config/axios";
import type { UserAchievementResponse } from "../types/achivement.ts";
import type { ApiResponse } from "../types/ApiResponse";

const API_BASE_URL = "/achievements";

export const achievementService = {
  getMyAchievements: async (): Promise<
    ApiResponse<UserAchievementResponse[]>
  > => {
    const response = await api.get(`${API_BASE_URL}/me`);
    return response.data;
  },

  getUserAchievements: async (
    userId: string,
  ): Promise<ApiResponse<UserAchievementResponse[]>> => {
    const response = await api.get(`${API_BASE_URL}/${userId}`);
    return response.data;
  },
};

export default achievementService;

File: mobile/lib/services/achievement_service.dart
import '../config/api_client.dart';
import '../models/achievement.dart';

class AchievementService {
  Future<UserDashboardResponse> getUserDashboard(String userId) async {
    final res = await apiClient.get('/users/$userId/dashboard');

    return UserDashboardResponse.fromJson(res.data['result']);
  }

  Future<List<UserAchievementResponse>> getUserAchievements(String userId) async {
    final res = await apiClient.get('/achievements/$userId');

    final List data = res.data['result'] ?? [];

    return data.map((e) => UserAchievementResponse.fromJson(e)).toList();
  }

  Future<List<UserAchievementResponse>> getMyAchievements() async {
    final res = await apiClient.get('/achievements/me');

    final List data = res.data['result'] ?? [];

    return data.map((e) => UserAchievementResponse.fromJson(e)).toList();
  }

}

final achievementService = AchievementService();